### PR TITLE
Generate a machine level certificate file that could be used later

### PR DESCRIPTION
**Pull request by bigbam505**
This change to use the machine level cert will be a breaking change for
any existing tentacles

### CODE REVIEW

File: libraries/shared.rb [hunk 1]
```diff
@@ -24,5 +24,17 @@ module Shared
     def fancy_bool(bool)
       bool.to_s.capitalize
     end
+
+    def actions_updated?(actions)
+      actions.each do |action|
+        return true if action.updated_by_last_action?
```
Comment (jonathanmorley) [inline libraries/shared.rb]:
```def actions_updated?(actions)
  actions.any? do |action|
    action.updated_by_last_action?
  end
end```

Comment (bigbam505) [inline libraries/shared.rb]: umm no, but....
```
actions.any?(&:updated_by_last_action?)
```

File: providers/tentacle.rb [hunk 2]
```diff
@@ -71,16 +72,30 @@
     version version
   end
 
-  configure = batch "configure-tentacle-#{instance}" do
+  generate_cert = powershell_script 'generate-tentacle-cert' do
+    cwd 'C:\Program Files\Octopus Deploy\Tentacle'
```
Comment (drapergeek) [inline providers/tentacle.rb]: What do you think about using a constant here or making this configureable?

Comment (bigbam505) [inline providers/tentacle.rb]: Done.

**Pull request merged**